code/datums/loadout/loadout_manager.dm

```diff
@@ -0,0 +1,88 @@
+/**
+ * This datum in charge with selecting wich loadout is currently being edited
+ * It also contains a tgui to navigate beetween loadouts
+ */
+/datum/loadout_manager
+	/// The loadout currently selected/modified
+	var/datum/loadout/current_loadout
+	/// A list of all loadouts
+	var/list/loadouts_list = list()
+	/// The data sent to tgui
+	var/loadouts_data = list()
+	/// The datum in charge of the user wanting to equip a saved loadout
+	var/datum/loadout_seller/seller
+
+/datum/loadout_manager/New()
+	seller = new
+
+///Remove a loadout from the list.
+/datum/loadout_manager/proc/delete_loadout(datum/loadout/loadout_to_delete)
+	loadouts_list -= loadout_to_delete
+	if(current_loadout == loadout_to_delete)
+		current_loadout = null
+	if(length(loadouts_data))
+		prepare_all_loadouts_data()
+
+///Prepare all loadouts data before sending them to tgui
+/datum/loadout_manager/proc/prepare_all_loadouts_data()
+	loadouts_data = list()
+	var/next_loadout_data = list()
+	for(var/datum/loadout/next_loadout AS in loadouts_list)
+		next_loadout_data = list()
+		next_loadout_data["job"] = next_loadout.job
+		next_loadout_data["name"] = next_loadout.name
+		loadouts_data += list(next_loadout_data)
+
+///Add one loadout to the loadout data
+/datum/loadout_manager/proc/add_loadout_data(datum/loadout/next_loadout)
+	var/next_loadout_data = list()
+	next_loadout_data["job"] = next_loadout.job
+	next_loadout_data["name"] = next_loadout.name
+	loadouts_data += list(next_loadout_data)
+
+/datum/loadout_manager/ui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		prepare_all_loadouts_data()
```

Comment:
why is this not purely in ui data

Reply:
ui data procs very often. 
Since i have bad experiences with static data (aka it's shit and never updated the comp when you force static data to update), i'm doing a custom system where everything is sent by ui_data, but the data sent is updated only when needed.
So it must be updated at start (so when creating the ui, so here), and when we are changing things (so in ui_act)

Reply:
can you not update ui_static_data on demand? I recall that there was a way

Reply:
Also could a malicious user spam this to lag the server